iroha/src/block_sync.rs

```diff
@@ -0,0 +1,180 @@
+use self::message::*;
+use crate::{
+    block::ValidBlock,
+    kura::Kura,
+    peer::PeerId,
+    sumeragi::{InitializedNetworkTopology, Role, Sumeragi},
+};
+use async_std::{sync::RwLock, task};
+use std::{sync::Arc, time::Duration};
+
+/// The state of `BlockSync`.
+#[derive(Clone)]
+pub enum State {
+    /// Not synchronizing now.
+    Idle,
+    /// Requested blocks from other peers.
+    Requested,
+    /// Synchronization is in progress: validating and committing blocks.
+    InProgress(Vec<ValidBlock>),
+}
+
+pub struct BlockSync {
```

Comment:
`Sync` means Synchronizer?

Reply:
Yes, do you think we should put a full name here?